src/core/trtserver.h

```diff
@@ -465,13 +466,103 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceManagerNew(
 TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceManagerDelete(
     TRTSERVER_TraceManager* trace_manager);
 
+/// TRTSERVER_InferenceRequestOptions
+///
+/// Object representing the request meta-data needed for an inference.
+///
+
+/// Create a new inference request options object.
+/// \param request_options Returns the new request options object.
+/// \param model_name The name of the model that the inference request is for.
+/// \param model_version The version of the model that the inference
+/// request is for, or -1 to select the latest (highest numbered) version.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestOptionsNew(
+    TRTSERVER_InferenceRequestOptions** request_options, const char* model_name,
+    int64_t model_version);
+
+/// Set the ID for the request in a request options. The response of the request
+/// will contain the same ID. The request sender can use the ID to correlate
+/// the response to corresponding request if needed.
```

Comment:
Add a sentence about what happens if this API is not called. I think the ID will be 0 in that case.